lib/RestructureCFG/InlineDivergentScopesPass.cpp

```diff
@@ -334,24 +376,123 @@ static void performIDS(DivergenceDescriptor &DivergenceDescriptor,
   simplifyTerminator(Tail, PlaceHolderTarget);
 
   // We add a `scope_closer` edge between the divergent exit node and
-  // the `Tail` node
-  ScopeGraphBuilder SGBuilder(F);
-  SGBuilder.addScopeCloser(Exit, Tail);
+  // the `Tail` node for all the exits
+  for (auto Divergence : MultipleDivergences) {
+    SGBuilder.addScopeCloser(Divergence.Exit, Tail);
+  }
+}
+
+/// Helper function which returns if a `DivergenceDescriptor` is compatible with
+/// all the ones already present in a `Collection`. Being compatible means that
+/// they insist on the same conditional, have non-overlapping `Exit` blocks, and
+/// do not have any overlapping `DivergentSuccessors`
+static bool
+canAddDescriptor(const SmallVector<DivergenceDescriptor> &Collection,
+                 const DivergenceDescriptor &NewDescriptor) {
```

Comment:
In general, it's clear what this function does, but it's not obvious the larger perspective of why you need to do this. What is the bad case you're guarding against?

Reply:
I changed the `canAddDescriptor` to `isCompatible`, and added a comment specifying that now this function is simply used as a failsafe check in an assertion here https://github.com/revng/revng/blob/1b4cabba417d846b3f4e47ca33b3f6b7b0a15234/lib/RestructureCFG/InlineDivergentScopesPass.cpp#L454

Reply:
:+1: